cuda/solver/common_trs_kernels.cuh

```diff
@@ -55,6 +55,94 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 namespace gko {
+namespace solver {
+
+
+struct SolveStruct {};
+
+
+namespace cuda {
+
+
+#if (defined(CUDA_VERSION) && (CUDA_VERSION >= 9020))
+
+
+struct SolveStruct : gko::solver::SolveStruct {
+    int algorithm;
+    csrsm2Info_t solve_info;
+    cusparseSolvePolicy_t policy;
+    cusparseMatDescr_t factor_descr;
+    size_t factor_work_size;
+    void *factor_work_vec;
+    SolveStruct()
+    {
+        factor_work_vec = nullptr;
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(cusparseCreateMatDescr(&factor_descr));
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(
+            cusparseSetMatIndexBase(factor_descr, CUSPARSE_INDEX_BASE_ZERO));
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(
+            cusparseSetMatType(factor_descr, CUSPARSE_MATRIX_TYPE_GENERAL));
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(
+            cusparseSetMatDiagType(factor_descr, CUSPARSE_DIAG_TYPE_NON_UNIT));
+        GKO_ASSERT_NO_CUSPARSE_ERRORS(cusparseCreateCsrsm2Info(&solve_info));
+        algorithm = 0;
+        policy = CUSPARSE_SOLVE_POLICY_USE_LEVEL;
+    }
+    SolveStruct(const SolveStruct &) : SolveStruct() {}
+    SolveStruct(SolveStruct &&) : SolveStruct() {}
+    SolveStruct &operator=(const SolveStruct &) { return *this; }
+    SolveStruct &operator=(SolveStruct &&) { return *this; }
```

Comment:
It seems to me like we only pass around pointers to SolveStruct. Could we ` = delete` these move/copy-constructors/assignments instead? This behavior is very unintuitive.